scala/scala_import.bzl

```diff
@@ -70,18 +77,29 @@ def _create_provider(
         for metadata in intellij_metadata:
             source_jars.extend(metadata.source_jars)
 
-    return java_common.create_provider(
-        use_ijar = False,
-        compile_time_jars = depset(
-            transitive = [current_target_compile_jars, exports.compile_jars],
-        ),
-        transitive_compile_time_jars = depset(transitive = [
-            jars.transitive_compile_jars,
-            current_target_compile_jars,
-            exports.transitive_compile_jars,
-        ]),
-        transitive_runtime_jars = depset(transitive = transitive_runtime_jars),
-        source_jars = source_jars,
+    deps = [
+        JavaInfo(
+            output_jar = jar,
+            compile_jar = jar
+        )
+        for jar in jars.transitive_compile_jars.to_list()
+    ]
+
+    deps.append([
+        JavaInfo(
+            output_jar = jar,
+            compile_jar = jar
+        )
+        for jar in transitive_runtime_jars
```

Comment:
should transitive *runtime* jars be added as direct compile time deps?

Reply:
I removed this method.